src/plugins/proctracer/proctracer.cpp

```diff
@@ -0,0 +1,401 @@
+        addr_t tmp_next = 0;
+        vmi_read_addr_va(vmi, next_module, pid, &tmp_next);
+
+        if (list_head == tmp_next)
+            break;
+
+        addr_t dllbase = 0;
+        vmi_read_addr_va(vmi, next_module + p->offsets[LDR_DATA_TABLE_ENTRY_DLLBASE], pid, &dllbase);
+
+        if (!dllbase){
+            traps_ok = false;
+            break;
+        }
+
+        unicode_string_t *us = vmi_read_unicode_str_va(vmi, next_module + p->offsets[LDR_DATA_TABLE_ENTRY_BASEDLLNAME], pid);
+        unicode_string_t out = { .contents = NULL };
+
+        if (us) {
+            status_t status = vmi_convert_str_encoding(us, &out, "UTF-8");
+            if(VMI_SUCCESS == status){
+                printf("Module name: %s\n", (char*)out.contents);
+                if (p->mod_config.find((char*)out.contents) != p->mod_config.end()){
+                    mod_info *mi = new mod_info;
+                    mi->mod_name = strdup((char*)out.contents);
+                    for (addr_t off: p->mod_config[(char*)out.contents]){
+                        trace_info *ti = new trace_info;
+                        drakvuf_trap_t *tracetrap = (drakvuf_trap_t*)g_malloc0(sizeof(drakvuf_trap_t));
+
+                        ti->mod_name = strdup((char*)out.contents);
+                        ti->offset = off;
+                        ti->p = p;
+
+                        addr_t trap_pa = vmi_pagetable_lookup(vmi, cr3, dllbase+off);
+                        tracetrap->breakpoint.lookup_type = LOOKUP_NONE;
+                        tracetrap->breakpoint.addr_type = ADDR_PA;
+                        tracetrap->type = BREAKPOINT;
+                        tracetrap->name = "TraceTrap";
+                        tracetrap->cb = trace_cb;
+                        tracetrap->breakpoint.addr = trap_pa;
+                        tracetrap->data = ti;
+
+                        if (drakvuf_add_trap(drakvuf,tracetrap)){
+                            p->trace_statistics[trap_pa]=0;
+                            mi->traps.push_back(tracetrap);
+                        }else{
+                            g_free(tracetrap);
+                            free(ti->mod_name);
+                            delete(ti);
+                        }
+                    }
+                    p->trace_status[cr3].push_back(mi);
+                }
+            }
+            vmi_free_unicode_str(us);
+        }
+
+        next_module = tmp_next;
+    }
+
+    return traps_ok;
+}
+
+// Runs on PsGetCurrentThreadTeb (process context)
+static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    proctracer *p = (proctracer*)info->trap->data;
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+
+    access_context_t ctx;
+    ctx.translate_mechanism = VMI_TM_PROCESS_DTB;
+    ctx.dtb = info->regs->cr3;
+
+    addr_t process = drakvuf_get_current_process(drakvuf, info->vcpu, info->regs);
+
+    if (!process){ 
+        drakvuf_release_vmi(drakvuf);
+        return -1;
```

Comment:
event_response_t has specific defined values that should be returned. Right now it's ignored by libdrakvuf but in the future it may get used, so returning 0 here is better.